test/libyul/yulOptimizerTests/rematerialiser/for_break.yul

```diff
@@ -0,0 +1,35 @@
+{
+	let a
+	let b
+	for {let i := 0} lt(i, 10) {i := add(a, b)} {
+		a := origin()
+		b := origin()
+
+		b := caller()
+		if callvalue() { break }
+		a := caller()
+	}
+}
```

Comment:
Actually we could do an `mstore(a, b)` here just to check that it is cleared after the loop.